### DISCUSSION TITLE
How to import json_project config programatically?

### DISCUSSION THREAD

**Discussion by cpinon-grd** [SCADA-LTS/Scada-LTS#2463] [Q&A] [answered]
Hello!

I am trying to dockerize Scada-LTS so that I can start it automatically from a config file with all my data sources/points (for instance, an exported json_project) but I haven't been able to find where is the config stored when I manually upload the JSON config in the UI import/export menu.

Could you provide some info about it?

Thank you in advance!

**Comment by Limraj** [comment, marked as answer]
Hi @cpinon-grd,
1. I don't know if this is what you're asking, but the 'Import project' function is to clean and update the mysql database, according to the data contained in the file. If you want to access this database, you can log into the container, then you can access it through SQL queries. Or put the database server port outside the container and connect with a client, e.g. Dbeaver.

2. However, if you have exported the configuration and you do not know where the file from this export is saved, it is saved locally according to the browser configuration, usually a Download folder is configured by default.

3. If you decide to use docker in a production environment, you need to configure: https://docs.docker.com/storage/volumes/. 

4. But If you are using linux ubuntu version < 22.04 (max version 20.04.5) or Linux Mint, then you can use the installer (without docker):
https://github.com/SCADA-LTS/installer

For more information, please contact: 
[scadalts-support@softq.pl](mailto:scadalts-support@softq.pl)

Best regards,
Kamil Jarmusik

**Reply by cpinon-grd** [reply]
Hello, Kamil, thanks for your reply.

What I am trying to accomplish is to use the exported configuration file as a config file so that when I launch another docker container, it reads this exported file and starts Scada-LTS with this config without having to manually upload this to the UI.

**Reply by Limraj** [reply]
It is important here to generate a cookie file that will be used in requests, and upload a configuration file that we want to import:
1. Login: 
curl -d "username=admin&password=admin&submit=Login" **-c cookies** http://localhost:8080/ScadaBR/login.htm
2. Upload file: (you have to replace **path_to_project_zip**, the @ character stays)
curl **-b cookies**  -v **-F importFile=@path_to_project_zip** http://localhost:8080/ScadaBR/import_project.htm
If ok:
![Screenshot 2023-01-24 at 10 15 07](https://user-images.githubusercontent.com/35842300/214253167-9991f4c2-33fe-4429-87a8-f3383bb61f64.png)
if not ok:
![Screenshot 2023-01-24 at 10 17 13](https://user-images.githubusercontent.com/35842300/214253528-41150072-52ae-4da2-916d-b8e0a6b8fd1f.png)

3. Import Project: 
curl 'http://localhost:8080/ScadaBR/dwr/call/plaincall/EmportDwr.loadProject.dwr' -X POST **-b cookies** --data-raw $'callCount=1\npage=/ScadaBR/import_project.htm\nhttpSessionId=\nscriptSessionId=D15BC242A0E69D4251D5585A07806324697\nc0-scriptName=EmportDwr\nc0-methodName=loadProject\nc0-id=0\nbatchId=5\n'

Be careful because the import delete all data from the database, and insert new data: 
- Therefore, if you send a request from step 3, the operation cannot be interrupted then.
- You also cannot do step 3 without doing step 2 within the same session.
- If you break something in the database, following these steps correctly gives you the opportunity to fix it.
Another way is to delete the entire database and create a new one, then the Application will reinitialize the structures in the database, but this will only restore the default structures and allow you to log in to the application as admin;

For export:
1. Login: 
curl -d "username=admin&password=admin&submit=Login" **-c cookies** http://localhost:8080/ScadaBR/login.htm
2. Export: 
curl -L **-b cookies** --output project.zip http://localhost:8080/ScadaBR/api/export/config/with-images/
